### DISCUSSION TITLE
8K video can't Download

### DISCUSSION THREAD

**Discussion by kjmdjul** [aclap-dev/video-downloadhelper#1496] [Help and Q&A]
### Your question

When I playing a 8K video and the resolution is showing about 4320p and when click on VDH there is showing 720p 1080p and up to 2160p and not showing for 4320p. So my asking how to download 4320p?

### Addon & CoApp version and details

![VDH 2](https://github.com/user-attachments/assets/504b552c-91ed-4b14-a92d-735fff78759b)
![VDH 1](https://github.com/user-attachments/assets/daa2e802-22de-472f-bc92-3d58dce355df)

**Comment by jcfrog** [comment]
Hi, I asked the dev team: "I believe the issue is the the point resolution mention in the video is a horizontal resolution. Whereas VDH shows the vertical resolution. Which is the standard way of doing it. If you download the 2160p VDH resolution you should get the 4320p resolution"

**Comment by kjmdjul** [comment]
Thank you so much jcfrog. But can I get refund for this purchase for
Chrome? If possible then plz tell me the way.

**Reply by jcfrog** [reply]
please send me a mail at licenses@downloadhelper.net

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

The answer from the "development team" appears to be entirely in error.  Look at this image very closely.

![#01](https://github.com/user-attachments/assets/e7322f8f-2fc5-440a-97dc-6616f6d88f5b)

The URL of that page is:

https://www.youtube.com/watch?v=gl3MzCdwOSQ&list=PLrnHu-uIjW-zVSb2EDukJDib4k_zUeqaP

Posters need to learn that in order to investigate problems properly, they need to post URLs, like I just did.  I think this is the video mentioned by the original poster.  It's the right duration & the thumbnail in the image above matches the initial image on the web page whose URL I just posted.  But the title appears to be somewhat different from what the original poster is showing.  So I'm not sure I've got the right video.  However, my comments below probably suffice to deal with this problem because this video has all the resolutions needed to exhibit the problem.

You'll note that there is a standard 2160p entry in the YouTube properties menu with the label 4K.  And then there's an even higher resolution entry of 4320p with the label 8K.  It is NOT the 3840x2160 variant.  The 3840x2160 variant is the one that VDH recognizes.  I downloaded it.  It's a standard 3840x2160 4K video.  VDH does not recognize the 8K variant.  The "development team" needs to actually investigate this, not dismiss it as a nonstandard way of labeling 4K content.  The 4K content is there.  IN ADDITION there is 8K content that VDH is NOT recognizing.

**Reply by GrampaWildWilly** [reply]
More evidence.  Here is a Network Monitor display for a different video.

![#02](https://github.com/user-attachments/assets/15da0cf7-834c-4174-bc34-8b83f9575dfc)

It is quite obvious here that 4320p is NOT a horizontal measure.  There really is content here that is 7680x4320.  The height, the VERTICAL measure, is 4320 pixels.  That makes it a 4320p variant.  This is not a mislabeling.  VDH is not recognizing this content.  This needs to be investigated as a legitimate problem report.

I would have tried to download this video but you can see the dreaded signatureCipher in front of the URL in the Network Monitor.  That is an indication that there is some secret incantation that needs to be performed on the URL before you can get something that can be downloaded.  That doesn't change the fact that VDH is failing to recognize this 8K content on YouTube.  There is a problem here.  It needs to be corrected.

**Comment by Mattmjs** [comment]
There may be 8K here but it's only an adaptive format, according to a tool I used there is no HLS version.
I think 8K is just not necessary anyway, our eyes don't perceive the extra detail beyond 4K. There are videos on that topic.

**Reply by GrampaWildWilly** [reply]
All the other resolutions, which VDH does recognize, are also labeled "adaptive" in that player json.  As for what our eyes are capable of perceiving is not the point.  The variant exists.  VDH should perceive it.  It does not.  That's a bug.

**Reply by Mattmjs** [reply]
I wouldn't say a bug, they made the decision to drop adaptive formats in favor of HLS since the adaptive formats are too problematic.
VDH doesn't detect adaptive formats anymore. I have another downloader for YouTube that gets it's downloads from those adaptive formats and it's been broken since early October, because Youtube changed something. VDH would have suffered the same thing had it stuck with those formats.

**Reply by GrampaWildWilly** [reply]
Oh sure, it's a bug.  It's not really even a new bug.  Something like this has happened before.  There was a short time during which VDH would not recognize any resolutions above 1080p.  They worked on it -- I believe "they" was Paul -- & now VDH recognizes 1440p & 2160p.  This is just another example of the same issue, just at a higher resolution.  Used to be that the VDH settings included this horrific list of resolutions.  It no longer does.  You can see this now via the `Export Settings` button in the VDH settings.  (That button does work.  There's just no feedback that it has worked.  It does save your settings in a json in your default `Download directory`.)  The old array of resolutions is now gone.  But it seems there still needs to be a tweak every time a new, probably higher, resolution is encountered in the wild.

As for HLS processing, Paul once told me that there is a way to poke & prod YouTube & it will spit out an HLS master manifest.  He seemed rather reluctant to explain exactly how that is done.  I suppose that's among the trade secrets that justify charging money for a VDH license.  It seems that the VDH CoApp uses this manifest to feed ffmpeg what it needs in order to download from YouTube.  So the issue of adaptive formats is a bit of a red herring.  If you have another tool that is broken at the moment, I'm sure it will eventually adapt, as will VDH.  The trick seems to be to catch Paul's attention & get him to actually work on this bug.  This is yet another in the long line of frivolous changes on YouTube that will make VDH jump through yet another hoop.  We just need to be patient.  I hope @jcfrog Jérôme can draw Paul's attention to this & get him to acknowledge this problem & work on it.

**Reply by jcfrog** [reply]
We're aware that the 8K version can be achieved in a somewhat convoluted way, but that's not really a subject we want to get into here. For your information, we're working on a new strategy for Youtube. "We just need to be patient" ;)
